src/storage/core.rs

```diff
@@ -133,8 +151,84 @@ impl<K: Key> Storage<K> {
         Ok(())
     }
 
+    /// Checks if there is a pending async operation
+    /// Returns boolean value (true - if there is, false otherwise)
+    /// Never falls
+    pub fn is_pending(&self) -> bool {
+        self.observer.is_pending()
+    }
+
+    /// FIXME: maybe it would be better to add check of `is_pending` state of observer for all
+    /// sync operations and return result in more appropriate way for that case (change Result<bool>
+    /// on Result<OpRes>, where OpRes = Pending|Done|NotDone for example)
+
+    /// Checks if active blob is set
+    /// Returns boolean value
+    /// Never falls
+    pub async fn has_active_blob(&self) -> bool {
+        self.inner.has_active_blob().await
+    }
+
+    /// Creates active blob
+    /// NOTICE! This function works in current thread, so it may take time. To perform this
+    /// asyncronously, use [`create_active_blob_non_blocking()`]
+    /// Returns true if new blob was created else false
+    /// # Errors
+    /// Fails if it's not possible to create new blob
+    /// [`create_active_blob_non_blocking()`]: struct.Storage.html#method.create_active_blob_async
+    pub async fn create_active_blob(&self) -> Result<bool> {
+        self.inner.create_active_blob().await
+    }
+
+    /// Creates active blob
+    /// NOTICE! This function returns immediately, so you can't check result of operation. If you
+    /// want be sure about operation's result, use [`create_active_blob()`]
+    /// [`create_active_blob()`]: struct.Storage.html#method.create_active_blob
+    pub async fn create_active_blob_non_blocking(&self) {

```

Comment:
Maybe it's better to rename this kind of fns:
`create_active_blob` -> `create_active_blob_spawned` or `create_active_blob_trigger` or `create_active_blob_in_background`
`create_active_blob_non_blocking` -> `create_active_blob`
Because it's kind of misleading for async fn to be `blocking`.

Reply:
It's a bit tricky, as `non_blocking` in this case refers to the function invocation and not to the underlying IO calls, and is a widespread term for this type of functions. Other terms can be misunderstood